src/test/java/StringManipulationTest.java

```diff
@@ -118,32 +156,45 @@ public void testRestoreString1()
         assertEquals(restoreString, "rat");
     }
 
+    //This test handles empty input values
     @Test
     public void testRestoreString2()
     {
-        fail("Not yet implemented");
-
+        manipulatedstring.setString("");
+        int [] array;
+        array=new int[]{};
+        String restoreString = manipulatedstring.restoreString(array);
+        assertEquals(restoreString, "");
     }
 
+    //This test handles duplicate values in a given index
     @Test
     public void testRestoreString3()
     {
-        fail("Not yet implemented");
-
+        manipulatedstring.setString("arty");
+        int [] array;
+        array=new int[]{1,0,1,2};
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.restoreString(array));
     }
 
+    //This test handles illegal values in the given index
     @Test
     public void testRestoreString4()
     {
-        fail("Not yet implemented");
-
+        manipulatedstring.setString("arty");
+        int [] array;
+        array=new int[]{-1,0,1,2};
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedstring.restoreString(array));
     }
 
+    //This test handles cases where s.length != indices.length
     @Test
     public void testRestoreString5()
     {
-        fail("Not yet implemented");
-
+        manipulatedstring.setString("art");
+        int [] array;
+        array=new int[]{1,0,2,3};
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.restoreString(array));
     }
 
```

Comment:
Your code handles illegalArgumentException in two situations; the first is when there are duplicates, and the second when the size of the array is either too big or too small relative to the length of the string. I see you testing the duplicates well, but what about different length of arrays being inputted? Another testRestoreString6 could be here.